tests/unit/test_probe.py

```diff
@@ -74,6 +76,27 @@ def value_err_raise():
             probe.join()
         assert (time.time() - start) < 1, "Timeout exceeded"
 
+    def test_count(self):
+        def say_no():
+            time.sleep(1)
+            return False
+
+        start = time.time()
+        probe = Probe(timeout=5, count=1, pause=0.5, fnc=say_no)
+        probe.run()
+        assert (time.time() - start) < 2, "Probe should end after one allowed try"
+
+        start = time.time()
+        probe = Probe(timeout=3, count=10, pause=0.5, fnc=say_no)
+        with pytest.raises(ProbeTimeout):
+            probe.run()
+        assert (time.time() - start) > 3, "Probe should reach timeout"
+
+        start = time.time()
+        probe = Probe(timeout=5, count=0, pause=0.5, fnc=value_err_raise)
```

Comment:
count=0 means don't run it?

Reply:
Right, this is just for testing purposes and it is not expected to be used like this.